plugins/catalog-react/src/alpha.tsx

```diff
@@ -37,21 +36,57 @@ export { useEntityPermission } from './hooks/useEntityPermission';
 export const entityContentTitleExtensionDataRef =
   createExtensionDataRef<string>('plugin.catalog.entity.content.title');
 
+/** @alpha */
+export const entityFilterExtensionDataRef = createExtensionDataRef<
+  (ctx: { entity: Entity }) => boolean
+>('plugin.catalog.entity.filter');
+
+function applyFilter(a?: string, b?: string): boolean {
+  if (!a) {
+    return true;
+  }
+  return a.toLocaleLowerCase('en-US') === b?.toLocaleLowerCase('en-US');
```

Comment:
Do you think typing errors might take time do catch? Adding a debug log would help in this case?